### PR TITLE
antimicro 2.3 -> antimicroX 3.0

### PR DESCRIPTION
**Pull request by KoviRobi**
Antimicro is broken an no longer maintained (and doesn't compile).
AntimicroX is a fork that does compile, so this removes antimicro and
adds antimicroX.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KoviRobi** [comment]
Sorry, I was meant to add a reviewer, not an assignee, and now can't remove you as an assignee.

**Comment by KoviRobi** [comment]
I have been using this software, because of working from home but having to use a work laptop, but my music is on my desktop computer. So I have hooked up a spare PS3 controller, and used this to map buttons to volume up/down, and play/pause. Seems to work fine.

**Review by jb55 — APPROVED** [review]
thanks! lgtm

File: pkgs/tools/misc/antimicroX/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    xorg.libX11
+    xorg.libXtst
+    xorg.libXi
+  ];
+
+  meta = with lib; {
+    description = "GUI for mapping keyboard and mouse controls to a gamepad";
+    inherit (src.meta) homepage;
+    maintainers = with maintainers; [ jb55 ];
+    license = licenses.gpl3;
```
Comment (danieldk) [inline pkgs/tools/misc/antimicroX/default.nix:41]: `gpl3` is deprecated, this should be `gpl3Only` or `gpl3Plus`. The source code has the *later* wording in the headers, so:

```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -25026,7 +25026,7 @@ in
 
   android-file-transfer = libsForQt5.callPackage ../tools/filesystems/android-file-transfer { };
 
-  antimicro = libsForQt5.callPackage ../tools/misc/antimicro { };
```
Comment (danieldk) [inline pkgs/top-level/all-packages.nix:25029]: Please add the `antimicro` attribute to `pkgs/top-level/aliases.nix` with a message pointing people to `antimicroX`. You can see the file for other examples.

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks for replacing a broken derivation! Some suggested improvements.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (KoviRobi) [reply pkgs/top-level/all-packages.nix:25029]: Thanks!

File: pkgs/tools/misc/antimicroX/default.nix (hunk 1 above)
Comment (KoviRobi) [reply pkgs/tools/misc/antimicroX/default.nix]: Thanks, done in the new commit!

File: pkgs/top-level/aliases.nix [hunk 3]
```diff
@@ -42,6 +42,7 @@ mapAliases ({
   aircrackng = aircrack-ng; # added 2016-01-14
   alienfx = throw "alienfx has been removed."; # added 2019-12-08
   ammonite-repl = ammonite; # added 2017-05-02
+  antimicro = throw "antimicro has been removed as it was broken, see antimicroX instead.";
```
Comment (danieldk) [inline pkgs/top-level/aliases.nix:45]: Sorry for nitpicking, could you also add the addition date, like the other entries?

Comment (KoviRobi) [reply pkgs/top-level/aliases.nix]: Yes of course, thank you for ensuring regularity :)

**Review by danieldk — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 91776` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>

  - antimicroX
</details>

Approved by package maintainer @jb55.

**Pull request merged**